### ISSUE TITLE
While doing fp16 training with AMPOptimizerCallback save GradScaler in CheckpointCallback

### ISSUE THREAD

**Issue by and-kul** [catalyst-team/catalyst#991]
## 🚀 Feature Request
As described in [PyTorch docs](https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html#saving-resuming) you must save not only `model`, `optimizer`, `criterion`, and `scheduler` but also `scaler` if you are doing fp16 training with automatic mixed precision.

### Motivation
Properly restore training state.

### Proposal
Probably we need to move `GradScaler` instance from `AMPOptimizerCallback` to runner  and update code of `CheckpointCallback`

PS
- [x] I know, that I could [join Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion.
